src/Analysis/GWASV2/GWASContainer.jsx

```diff
@@ -1,110 +1,131 @@
-import "./GWASV2.css";
+import React, { useState } from 'react';
+import {
+  Space, Button, Popconfirm, Spin,
+} from 'antd';
+import SelectStudyPopulation from './SelectStudyPopulation/SelectStudyPopulation';
+import SelectOutcome from './SelectOutcome/SelectOutcome';
+import SelectCovariates from './SelectCovariates/SelectCovariates';
+import './GWASV2.css';
+import { gwasSteps } from './constants';
+import { useSourceFetch } from '../GWASWizard/wizardEndpoints/cohortMiddlewareApi';
 
 const GWASContainer = () => {
   const [current, setCurrent] = useState(0);
   const [
     selectedStudyPopulationCohort,
     setSelectedStudyPopulationCohort,
   ] = useState({});
-  const gwasSteps = [
-    {
-      title: "Step 1",
-      description: "Select Study Population",
-    },
-    {
-      title: "Step 2",
-      description: "Select Outcome Phenotypes",
-    },
-    {
-      title: "Step 3",
-      description: "Select Covariate Phenotype",
-    },
-    {
-      title: "Step 4",
-      description: "Configure GWAS",
-    },
-  ];
+  const [allCovariates] = useState([]);
+  const [outcome, setOutcome] = useState({});
+  const { loading, sourceId } = useSourceFetch();
+  const [selectedCovariate, setSelectedCovariate] = useState(undefined);
+
+  const handleCovariateSubmit = (cov) => {
+    setSelectedCovariate(cov);
+  };
+
+  const handleOutcome = (out) => {
+    setOutcome(out);
+  };
 
   const generateStep = () => {
     // steps 2 & 3 very similar
     switch (current) {
-      case 0:
-        // select study population
-        return (
-          <SelectStudyPopulation
-            selectedStudyPopulationCohort={selectedStudyPopulationCohort}
-            setSelectedStudyPopulationCohort={setSelectedStudyPopulationCohort}
-            current={current}
-          />
-        );
-      case 1:
-        // outcome (customdichotomous or not)
-        return <React.Fragment>step 2</React.Fragment>;
-      case 2:
-        // covariates (customdichtomous or not)
-        return <React.Fragment>step 3</React.Fragment>;
-      case 3:
-        // all other input (mafs, imputation, etc), review, and submit
-        return <React.Fragment>step 4</React.Fragment>;
-      default:
-        // required for eslint
-        return null;
+    case 0:
+      // select study population
+      return (
+        <SelectStudyPopulation
+          selectedStudyPopulationCohort={selectedStudyPopulationCohort}
+          setSelectedStudyPopulationCohort={setSelectedStudyPopulationCohort}
+          current={current}
+          sourceId={sourceId}
+        />
+      );
+    case 1:
+      // outcome (customdichotomous or not)
+      return (
+        <SelectOutcome
+          outcome={outcome}
+          // allCovariates={allCovariates}
+          handleOutcome={handleOutcome}
+          sourceId={sourceId}
+          current={current}
+        />
+      );
+    case 2:
+      // covariates (customdichtomous or not)
+      return (
```

Comment:
this step is about _outcome_ phenotype (selecting covariates is only done in next step). I propose we change the naming to reflect this as much as possible.